lib/game.rb

```diff
@@ -0,0 +1,64 @@
+class Game
+  attr_accessor :board, :player_1, :player_2, :cells
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [6,4,2]
+  ]
+
+  def initialize(player_1 = Player::Human.new("X"), player_2 = Player::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def won?
+    WIN_COMBINATIONS.detect do |combo|
+      combo.all? { |position| board.cells[position] == player_1.token } ||
+      combo.all? { |position| board.cells[position] == player_2.token }
+    end
+  end
+
+  def draw?
+    !won? && board.full?
+  end
+
+  def winner
+    board.cells[won?.first] if won?
```

Comment:
As written, your `winner` method will execute `won?` twice - first when the `if` statement is evaluated. and second when selecting the index in `cells` using `won?.first`. The issue with this relates to performance, in that `won?` has to iterate through all of the WIN_COMBINATIONS.

To avoid this, you can capture the return value of `won?` and store it in `won`. If `won?` returns a value, `won = won?` will evaluate to that return value, which is 'truthy'. On the other hand, if `won?` fails to return a value, because no winning combination was detected, it will return `nil`; therefore the statement `won = won?` will also return `nil`, which is 'falsey'.

So, this revised approach is saying, if `won?` detects a winner and returns a value, pass that to board.cells to retrieve the winner's token; otherwise, return `nil`.